vol_object_test.c

```diff
@@ -5115,25 +5134,72 @@ test_object_visit(void)
         goto error;
     }
 
-    if ((fspace_id = generate_random_dataspace(OBJECT_VISIT_TEST_SPACE_RANK, NULL, NULL, FALSE)) < 0)
-        TEST_ERROR;
+    /* Make sure not to generate too much data for an attribute to hold */
+    do {
+        if (fspace_id != H5I_INVALID_HID)
+            H5Sclose(fspace_id);
```

Comment:
probably need to set `fspace_id` and `dset_dtype` back to `H5I_INVALID_HID` after the closes since the close operation itself won't set them to that.